Fix NJT frexp() to handle both outputs

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #144889
* #144588
* #144587
* #144586
* __->__ #144585
* #144584
* #144583
* #144582

Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

Before this PR, `frexp()` for NJT was handled via the unary pointwise fallback. The op returns a tuple, however, and the fallback doesn't handle that. This PR defines an explicit impl for `frexp()` that wraps both returned `(mantissa, exponent)` as NJTs.

jbschlosser committed:
Fix NJT frexp() to handle both outputs

[ghstack-poisoned]

Skylion007 commented on torch/nested/_internal/ops.py:
```diff
@@ -2466,6 +2466,21 @@ def fill__Scalar(func, *args, **kwargs):
 
 
+@register_jagged_func(torch.ops.aten.frexp.Tensor, "self: jt_all")
+def frexp_Tensor(func, *args, **kwargs):
```
NIt: can we type frexp_Tensor with ParamSpec?

jbschlosser committed:
Update on "Fix NJT frexp() to handle both outputs"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

Before this PR, `frexp()` for NJT was handled via the unary pointwise fallback. The op returns a tuple, however, and the fallback doesn't handle that. This PR defines an explicit impl for `frexp()` that wraps both returned `(mantissa, exponent)` as NJTs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Fix NJT frexp() to handle both outputs"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

Before this PR, `frexp()` for NJT was handled via the unary pointwise fallback. The op returns a tuple, however, and the fallback doesn't handle that. This PR defines an explicit impl for `frexp()` that wraps both returned `(mantissa, exponent)` as NJTs.

[ghstack-poisoned]

jbschlosser committed:
Update on "Fix NJT frexp() to handle both outputs"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

Before this PR, `frexp()` for NJT was handled via the unary pointwise fallback. The op returns a tuple, however, and the fallback doesn't handle that. This PR defines an explicit impl for `frexp()` that wraps both returned `(mantissa, exponent)` as NJTs.

[ghstack-poisoned]

jbschlosser commented:
@pytorchbot merge

jbschlosser committed:
Update on "Fix NJT frexp() to handle both outputs"


Part of my BE project addressing NJT bugs surfaced via OpInfo tests.

Before this PR, `frexp()` for NJT was handled via the unary pointwise fallback. The op returns a tuple, however, and the fallback doesn't handle that. This PR defines an explicit impl for `frexp()` that wraps both returned `(mantissa, exponent)` as NJTs.

[ghstack-poisoned]

Skylion007 commented:
@pytorchbot merge
